### PR TITLE
first implementation of validating node names

### PR DESCRIPTION
**Pull request by zschutzman**
Migrating discussion of Issue #280 (making sure assignments have nodes named in the same way as the graph).

Points to discuss: 

- Is the initialization of the `Partition` object the right place to do this check?
- Should the `Assignment` class implement the `collections.Mapping` stuff?
- Is `NameError` the right exception? `KeyError`? `ValueError`?

### CODE REVIEW

File: gerrychain/partition/partition.py [hunk 1]
```diff
@@ -51,6 +55,11 @@ def _first_time(self, graph, assignment, updaters):
         self.flows = None
         self.edge_flows = None
 
+    def validate_assignment(self):
+        node_names = set(list(self.graph.nodes()))
```
Comment (maxhully) [inline gerrychain/partition/partition.py]: I think we can shorten this to:
```suggestion
        node_names = set(self.graph.nodes)
```

File: gerrychain/partition/partition.py [hunk 2]
```diff
@@ -51,6 +55,11 @@ def _first_time(self, graph, assignment, updaters):
...
+        assgn_names = set([name for dist in self.assignment.parts.values() for name in dist])
```
Comment (maxhully) [inline gerrychain/partition/partition.py]: The `set()` constructor can take the plain generator expression instead of the list:
```suggestion
        assgn_names = set(name for dist in self.assignment.parts.values() for name in dist)
```

File: gerrychain/partition/partition.py [hunk 3]
```diff
@@ -41,6 +41,10 @@ def _first_time(self, graph, assignment, updaters):
 
         self.assignment = get_assignment(assignment, graph)
 
+         if not self.validate_assignment():
```
Comment (maxhully) [inline gerrychain/partition/partition.py]: Looks like there's an extra space before this `if` that's causing the failing tests. You can fix it manually or outsource it to a formatter like [black](https://black.readthedocs.io/en/stable/).

Comment (zschutzman) [reply gerrychain/partition/partition.py]: edited manually

**Comment by zschutzman** [comment]
I've committed the changes and fixed the formatting problem.  If `NameError` is okay, then I think it's good to go?
